EventFilter/SiPixelRawToDigi/interface/PixelDataFormatter.h

```diff
@@ -107,6 +107,8 @@ class PixelDataFormatter {
   int maxROCIndex;
   bool phase1;
 
+  friend class SiPixelRawToDigi;
```

Comment:
it looks like this is added only to get access to LINK_shift and LINK_mask.
Please add accessors to these or better yet define a ```linkId(const SiPixelRawDataError& error) const``` to encapsulate it fully in a class/method that actually knows what's inside

Reply:
PixelFEDChannel is a generic class that can be used to list pixel channels in any code, should not even need to know what FED errors are.
Error code 25 is a firmware choice; in general, I would not find it advisable to use it in any function/class names.
Surely, this PR  hard-codes this error in the unpacker, but at least it is done explicitly, localized in a place that is the most obvious choice (where we are filling special collections derived from the generic FED error collection) for any pixel experts.
Note, there is not a single bit-field decoding function in PixelDataFormatter. Defining one would mean that it should be used exclusively everywhere. Points outside the scope of this PR.

Reply:
> there is not a single bit-field decoding function in PixelDataFormatter

Every single function in PixelDataFormatter does bit-field operations either on the argument passed to it directly or on something derived from it.
And prior to this PR this was the only place where such manipulations were done.
It would be better to keep it that way and not expose bitmasks and shifts to outside functions.
[I'm just restating the points of previous comments]